src/models/decoder_only.cpp

```diff
@@ -50,4 +85,10 @@ void DecoderOnly_State::UpdateInputsOutputs(DeviceSpan<int32_t>& next_tokens, De
   logits_.Update(next_tokens, new_length);
 }
 
+bool DecoderOnly_State::IsChunkingSupported() const {
+  // Enable prefill chunking for CUDA and NvTensorRtRtx devices
+  return (model_.p_device_->GetType() == DeviceType::CUDA ||
+          model_.p_device_->GetType() == DeviceType::NvTensorRtRtx);
```

Comment:
Let's avoid using this device check. Instead let's check if the chunking is enabled through the config alone. It should be a sufficient check to avoid doing chunked prefill.